modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -2912,8 +2869,8 @@ void GDScriptAnalyzer::reduce_identifier_from_base(GDScriptParser::IdentifierNod
 						case GDScriptParser::ClassNode::Member::CONSTANT: {
 							// TODO: Make sure loops won't cause problem. And make special error message for those.
 							// For out-of-order resolution:
-							reduce_expression(member.constant->initializer);
-							p_identifier->set_datatype(member.get_datatype());
+							resolve_constant(member.constant);
```

Comment:
yessssssssssssssssssssss

Reply:
Thanks) 

I do think that other places like `resolve_variable` (with it's code copy in `resolve_class_interface`) and `resolve_parameter` can be unified with `resolve_constant` in the future under the same function too.

Reply:
Yes, there's a lot of refactoring possible here, and I for one am excited for it :)